File: Java/백준/Gold/1101. 카드 정리 1/카드 정리 1.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt(); // 박스 수
        int M = sc.nextInt(); // 색상 수
        int[][] arr = new int[N][M];

        for (int i = 0; i < N; i++)
            for (int j = 0; j < M; j++)
                arr[i][j] = sc.nextInt();

        int ans = N - 1; // 최악의 경우 N-1번 이동

        for (int joker = -1; joker < N; joker++) {
            boolean[] used = new boolean[M];
            int cnt = 0;

            for (int i = 0; i < N; i++) {
                if (i == joker) continue; // 조커 박스는 건너뜀

                int flag = 0;
                for (int j = 0; j < M; j++) {
                    if (arr[i][j] != 0) flag++;
                }

                if (flag == 0) continue; // 비어있으면 이동 X

                if (flag == 1) {
                    int color = -1;
                    for (int j = 0; j < M; j++) {
                        if (arr[i][j] != 0) {
                            color = j;
                            break;
                        }
                    }
                    if (used[color]) {
                        cnt++; // 이미 쓰인 색이면 이동 필요
                    } else {
                        used[color] = true;
                    }
                } else {
                    cnt++; // 두 색 이상이면 무조건 이동
                }
            }

            ans = Math.min(ans, cnt);
        }

        System.out.println(ans);
    }
}

File: C++17/백준/Gold/1101. 카드 정리 1/카드 정리 1.cc
#include <iostream>
#include <cstring>

using namespace std;

int main() {    
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);

    int N,M,i,j,arr[50][50],joker,ans=0,cnt;
    bool used[50] = {};

    cin >> N >> M;
    ans = N-1;
    for(i=0;i<N;i++) {
        for(j=0;j<M;j++) {
            cin >> arr[i][j];
        }
    }

    for(joker=0;joker<N;joker++) {
        memset(used,false,sizeof(used));
        cnt = 0;
        for(i=0;i<N;i++) {
            if(joker == i)
                continue;
            else {
                int flag = 0;  // check number of not 0
                for(j=0;j<M;j++) {
                    if(arr[i][j] != 0) {
                        flag++;
                    }
                }
                if(!flag) // all 0 -> dont have to move
                    continue;
                if(flag == 1) { // only one color
                    int color;
                    for(j=0;j<M;j++) // find which color
                        if(arr[i][j])
                            color = j;
                    if(used[color]) // already used color
                        cnt++;
                    else // first time
                        used[color] = true;
                } else if(flag>1) { // over two colors, have to move
                    cnt++;
                }
            }
        }
        ans = min(ans, cnt);
    }
    cout << ans << "\n";
    return 0;
}